src/uniformscalingmap.jl

```diff
@@ -58,41 +60,27 @@ end
 
 function _scaling!(y, λ, x, α, β)
     if (iszero(α) || iszero(λ))
-        iszero(β) && (fill!(y, zero(eltype(y))); return y)
+        iszero(β) && return fill!(y, zero(eltype(y)))
         isone(β) && return y
-        # β != 0, 1
-        rmul!(y, β)
-        return y
+        return rmul!(y, β)
+    elseif isone(α) && isone(λ)
+        iszero(β) && return copyto!(y, x)
+        isone(β) && return y .+= x
+        return y .= y .* β .+ x
     elseif isone(α)
-        if iszero(β)
-            isone(λ) && return copyto!(y, x)
-            y .= λ .* x
-            return y
-        elseif isone(β)
-            isone(λ) && return y .+= x
-            y .+= λ .* x
-            return y
-        else # β != 0, 1
-            isone(λ) && (axpby!(one(eltype(x)), x, β, y); return y)
-            y .= y .* β .+ λ .* x
-            return y
-        end
-    else # α != 0, 1
-        if iszero(β)
-            isone(λ) && (y .= x .* α; return y)
-            y .= λ .* x .* α
-            return y
-        elseif isone(β)
-            isone(λ) && (axpby!(α, x, β, y); return y)
-            y .+= λ .* x .* α
-            return y
-        else # β != 0, 1
-            isone(λ) && (y .= y .* β .+ x .* α; return y)
-            y .= y .* β .+ λ .* x .* α
-            return y
-        end # β-cases
-    end # α-cases
-end # function _scaling!
+        iszero(β) && return y .= λ .* x
+        isone(β) && return y .+= λ .* x
+        return y .= y .* β .+ λ .* x
+    elseif isone(λ)
+        iszero(β) && return y .= x .* α
+        isone(β) && return y .+= x .* α
+        return y .= y .* β .+ x .* α
+    else
+        iszero(β) && return y .= λ .* x .* α
+        isone(β) && return y .+= λ .* x .* α
+        return y .= y .* β .+ λ .* x .* α
+    end
+end
```

Comment:
This was originally so "complicated", because we wanted to save the multiplication by `λ` in case it's 1. But you don't seem to think that's relevant? I'm fine with that, just wanted to make sure it has been said and considered.

Reply:
The `isone(λ)` case is handled above no?

Reply:
You're right. All the lines basically look so similar, so I overlooked that. 😅